sharding/nodesCoordinator/indexHashedNodesCoordinatorLite_test.go

```diff
@@ -4,86 +4,163 @@ import (
+			TempRating: 10},
+		{
+			PublicKey:  []uint8{1, 2, 3, 5},
+			ShardId:    core.MetachainShardId,
+			List:       "eligible",
+			Index:      2,
+			TempRating: 10},
+	}
+}
+
 func TestIndexHashedNodesCoordinator_SetNodesConfigFromValidatorsInfo(t *testing.T) {
 	t.Parallel()
 
 	arguments := createArguments()
 
-	ihgs, err := NewIndexHashedNodesCoordinator(arguments)
+	shufflerArgs := &NodesShufflerArgs{
+		NodesShard:           3,
+		NodesMeta:            3,
+		Hysteresis:           hysteresis,
+		Adaptivity:           adaptivity,
+		ShuffleBetweenShards: shuffleBetweenShards,
+		MaxNodesEnableConfig: nil,
+	}
+	nodeShuffler, _ := NewHashValidatorsShuffler(shufflerArgs)
+	arguments.Shuffler = nodeShuffler
+
+	ihnc, err := NewIndexHashedNodesCoordinator(arguments)
 	require.Nil(t, err)
 	epoch := uint32(1)
 
-	randomness := []byte("rand seed")
-
-	ihgs.nodesConfig[epoch] = ihgs.nodesConfig[0]
+	validatorsInfo := createDummyValidatorsInfo()
 
-	body := createBlockBodyFromNodesCoordinator(ihgs, epoch)
-	validatorsInfo, err := createValidatorInfoFromBody(body, arguments.Marshalizer, 10)
+	err = ihnc.SetNodesConfigFromValidatorsInfo(epoch, []byte("rand seed"), validatorsInfo)
 	require.Nil(t, err)
 
-	err = ihgs.SetNodesConfigFromValidatorsInfo(epoch, randomness, validatorsInfo)
-
-	validators, err := ihgs.GetAllEligibleValidatorsPublicKeys(epoch)
-	require.Nil(t, err)
-	require.NotNil(t, validators)
+	verifyValidatorsPubKeysByIndex(t, validatorsInfo, ihnc.nodesConfig[epoch])
 }
 
 func TestIndexHashedNodesCoordinator_SetNodesConfigFromValidatorsInfoMultipleEpochs(t *testing.T) {
 	t.Parallel()
 
 	arguments := createArguments()
 
-	ihgs, err := NewIndexHashedNodesCoordinator(arguments)
+	shufflerArgs := &NodesShufflerArgs{
+		NodesShard:           3,
+		NodesMeta:            3,
+		Hysteresis:           hysteresis,
+		Adaptivity:           adaptivity,
+		ShuffleBetweenShards: shuffleBetweenShards,
+		MaxNodesEnableConfig: nil,
+	}
+	nodeShuffler, _ := NewHashValidatorsShuffler(shufflerArgs)
+	arguments.Shuffler = nodeShuffler
+
+	ihnc, err := NewIndexHashedNodesCoordinator(arguments)
 	require.Nil(t, err)
 	epoch := uint32(1)
 
 	randomness := []byte("rand seed")
 
-	ihgs.nodesConfig[epoch] = ihgs.nodesConfig[0]
+	validatorsInfo := createDummyValidatorsInfo()
 
-	body := createBlockBodyFromNodesCoordinator(ihgs, epoch)
-	validatorsInfo, err := createValidatorInfoFromBody(body, arguments.Marshalizer, 10)
+	err = ihnc.SetNodesConfigFromValidatorsInfo(epoch, randomness, validatorsInfo)
+	require.Nil(t, err)
+	err = ihnc.SetNodesConfigFromValidatorsInfo(epoch+1, randomness, validatorsInfo)
 	require.Nil(t, err)
 
-	ihgs.SetNodesConfigFromValidatorsInfo(epoch, randomness, validatorsInfo)
-	ihgs.SetNodesConfigFromValidatorsInfo(epoch+1, randomness, validatorsInfo)
-
-	_, ok := ihgs.nodesConfig[epoch+1]
+	_, ok := ihnc.nodesConfig[epoch+1]
 	assert.True(t, ok)
+	verifyValidatorsPubKeysByIndex(t, validatorsInfo, ihnc.nodesConfig[epoch+1])
```

Comment:
```suggestion
	epochConfig, ok := ihnc.nodesConfig[epoch+1]
	assert.True(t, ok)
	verifyValidatorsPubKeysByIndex(t, validatorsInfo, epochConfig)
```

Applies globally here